dropshot/src/schema_util.rs

```diff
@@ -202,38 +203,35 @@ fn schema2struct_impl(
 
 #[derive(Debug)]
 pub(crate) enum Schema2StructError {
+    InvalidEnum(schemars::schema::Schema),
     InvalidType(schemars::schema::Schema),
     InvalidSubschema(schemars::schema::SubschemaValidation),
-    UnitEnum(schemars::schema::Schema),
-    ComplexEnum(schemars::schema::Schema),
 }
 
 impl fmt::Display for Schema2StructError {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            Schema2StructError::InvalidType(schema) => {
-                write!(f, "invalid type {:#?}", schema)
-            }
-            Schema2StructError::InvalidSubschema(subschemas) => {
-                write!(f, "invalid subschema {:#?}", subschemas)
+            Schema2StructError::InvalidEnum(schema) => {
+                write!(
+                    f,
+                    "invalid type: {}\n\
+                    (hint: this appears to be an enum, \
+                     which needs to be wrapped in a struct)",
+                    serde_json::to_string_pretty(schema).unwrap(),
+                )
             }
-            Schema2StructError::UnitEnum(schema) => {
+            Schema2StructError::InvalidType(schema) => {
                 write!(
                     f,
-                    "invalid type {:#?}\n\
-                     (hint: this appears to be an enum with unit variants, \
-                     and needs to be wrapped in a struct)",
-                    schema
+                    "invalid type: {}",
+                    serde_json::to_string_pretty(schema).unwrap(),
```

Comment:
A change I snuck in was to use `serde_json::to_string_pretty`, as Adam recommended in chat.